### ISSUE TITLE
Datasource password property causing SSMS generation of XMLA script to fail

### ISSUE THREAD

**Issue by alwortman** [TabularEditor/TabularEditor3#210]
Our existing deployment pipelines use the Microsoft.AnalysisServices.Deployment executable to generate XMLA scripts which then get deployed via Powershell (Invoke-ASCmd). Since switching over to TE3, we've noticed some (but seemingly not all) models have a password property saved in the model definition. Here is an example, in this case the password property is _literally_ the asterisks as shown below:

```
{
  "name": "WorkspaceDB_Factoryjmiranda_20210809",
  "compatibilityLevel": 1500,
  "model": {
    "name": "Hephaestus",
    "culture": "en-US",
    "dataSources": [
      {
        "type": "structured",
        "name": "Enterprise HANA",
        "connectionDetails": {
          "protocol": "sap-hana-sql",
          "address": {
            "server": "sapehpdb.intel.com:31015"
          },
          "authentication": null,
          "query": null
        },
        "options": {
          "implementation": "2.0"
        },
        "credential": {
          "AuthenticationKind": "Windows",
          "Username": "amr\\jmiranda",
          "EncryptConnection": false,
          "PrivacySetting": "Public",
          "Password": "********"
        }
      },
```

I _can_ deploy using Microsoft.AnalysisServices.Deployment in silent mode if I don't generate the XMLA script (deploy to the server directly) or if I generate a TMSL script with interactive mode, then deploy.

The error I get is:
`Unexpected end of content while loading JObject. Path 'PrivacySetting', line 5, position 29.`

- TE3 Version: 3.0.10
- SSMS Version 18.9.2

**Comment by otykier** [comment]
From your post it sounds like we're dealing with two separate issues:

1) Tabular Editor stores the literal asterisks in the password property (where as actually, it shouldn't serialize this property at all).
2) An error when executing a TMSL script generated through Tabular Editor. I didn't quite get where this error originates from? Was it when executing through SSMS or InvokeASCmd? What happens if you manually remove `"PrivacySetting": "Public"` from the TMSL script?

**Comment by alwortman** [comment]
@otykier I think you are correct.

The error surfaces when trying to execute this command line argument (from within the folder with the .asdatabase, .deploymenttargets files, etc.):
```"C:\Program Files (x86)\Microsoft SQL Server Management Studio 18\Common7\IDE\Microsoft.AnalysisServices.Deployment.exe" /s:log.txt /o:Model.xmla```

That generates a message in the log file that looks like this:
```
Connecting to the FM7MSBIDEVTAB6\SSAS2019 server
Database, FAE_Factory_Latest, found on server, FM7MSBIDEVTAB6\SSAS2019. Applying configuration settings and options...
	Analyzing configuration settings...
	Done
	Analyzing storage information...
	Done
	Analyzing security information...
		Analyzing permissions...
		Admin Role
		User Role
		FAE Olympus Role
		DARC Team Role
		Done
	Done
Generating processing sequence...
Saving deployment script to C:\agent\_work\r17\a\_Hephaestus-Main-CI\drop\Model.xmla
Unexpected end of content while loading JObject. Path 'PrivacySetting', line 5, position 29.
```

**Comment by otykier** [comment]
This is fixed in [3.1.1](https://docs.tabulareditor.com/te3/other/release-notes/3_1_1.html). 

There is a new option under **Model > Serialization options... > Ignore Privacy Setting** which should be checked to make sure that the `PrivacySetting` property is not saved in the .bim file, which should solve the issue with Microsoft.AnalysisServices.Deployment.

Thanks for reporting!
